modules/core/src/test/java/org/apache/ignite/testframework/junits/IgniteConfigVariationsAbstractTest.java

```diff
@@ -108,9 +92,17 @@ protected static void injectTestsConfiguration(VariationsTestsConfig testsCfgInj
         return false;
     }
 
+    /** Check that test name is not null. */
+    @Before
+    public void checkTestName(){
+        assert getName() != null : "getName returned null";
+    }
+
     /** {@inheritDoc} */
     @Override protected void beforeTestsStarted() throws Exception {
-        assert testsCfg != null;
+        assert testsCfgInjected != null;
+
+        testsCfg = testsCfgInjected;
```

Comment:
Now `testsCfg` is assigned before all tests in a class. Before that change it was assigned before each test method execution. Could you please explain why it still works as expected?

Reply:
Ivan, as I see testsCfg do not change during test methods execution. So the reason for me was to remove unnecessary initialization.
But I can try to move such initialization to beforeTest() method in IgniteConfigVariationsAbstractTest class to leave the same logic as it was before.

Reply:
I mean that we inject TestConfigurations only once [1] and I do not see a reason to re-initialize testCfg before each method.
[1] https://github.com/apache/ignite/blob/master/modules/core/src/test/java/org/apache/ignite/testframework/test/ConfigVariationsTestSuiteBuilderTest.java#L401

Reply:
@1vanan I suppose it is expected to work as follows:
1. Test classes are created using a code generation library.
2. `testsCfgInjected` (static member) is assigned on `@BeforeClass` level.
3. `testsCfg` (instance member) is assigned on `@Before` level.

A test class instance is created for each test method execution. So, I do not understand how an expected configuration is injected to each test instance. Could you please that configurations are injected as expected? I suspect that `dummyCfg` is injected everywhere now.

Reply:
@pavlukhin you are definitely right.
Please check the last commit - I inserted [1] debug info for checking testCfg. When we assign testsCfgInjected before test class, it returns "Dummy config", before each test - another cfg description. 
So I think to remain the last version (in the last commit) - in beforeTest method.
[1] https://github.com/apache/ignite/pull/6434/files#diff-78ef107e580c29ced70d7eb0739f2c7bR147